dataprofiler/tests/data_readers/test_data.py

```diff
@@ -160,6 +161,23 @@ def test_read_url_verify_ssl(self, mock_request_get):
         ):
             data_obj = Data("https://test.com")
 
+    @patch("boto3.client")
+    @patch(
+        "os.environ",
+        {
+            "AWS_ACCESS_KEY_ID": "<YOUR_ACCESS_KEY>",
+            "AWS_SECRET_ACCESS_KEY": "<YOUR_SECRET_KEY>",
+        },
+    )
+    def test_read_s3_uri(self, mock_boto3_client):
```

Comment:
we should validate that `mock_boto3_client` is called with particular values. Something like this below 

```
            mock_boto3_client.assert_called_with(
                "s3",
                aws_access_key_id="<YOUR_ACCESS_KEY>",
                aws_secret_access_key="<YOUR_SECRET_KEY>",
                aws_session_token=None,
                region_name='us-east-1',
            )
```

Reply:
Agreed. I had already the same call on `test_s3_connection.py`.  Will add it here too.